crates/network/src/discovery/state.rs

```diff
@@ -1,90 +1,98 @@
-use std::collections::{btree_map, BTreeMap, BTreeSet, HashSet};
-use std::time;
+#[cfg(test)]
+#[path = "../tests/discovery/state.rs"]
+mod tests;
+
+use core::time::Duration;
+use std::collections::btree_map::Entry;
+use std::collections::{BTreeMap, BTreeSet, HashSet};
+use std::time::Instant;
 
-use libp2p::{rendezvous, Multiaddr, PeerId, StreamProtocol};
+use libp2p::relay::HOP_PROTOCOL_NAME;
+use libp2p::rendezvous::Cookie;
+use libp2p::{Multiaddr, PeerId, StreamProtocol};
+use multiaddr::Protocol;
 
 // The rendezvous protocol name is not public in libp2p, so we have to define it here.
 // source: https://github.com/libp2p/rust-libp2p/blob/a8888a7978f08ec9b8762207bf166193bf312b94/protocols/rendezvous/src/lib.rs#L50C12-L50C92
-const RENDEZVOUS_PROTOCOL_NAME: libp2p::StreamProtocol =
-    libp2p::StreamProtocol::new("/rendezvous/1.0.0");
+const RENDEZVOUS_PROTOCOL_NAME: StreamProtocol = StreamProtocol::new("/rendezvous/1.0.0");
 
 /// DiscoveryState is a struct that holds the state of the disovered peers.
 /// It holds the relay and rendezvous indexes to quickly check if a peer is a relay or rendezvous.
 /// It offers mutable methods for managing the state of the peers.
 #[derive(Debug, Default)]
-pub(crate) struct DiscoveryState {
+pub struct DiscoveryState {
     peers: BTreeMap<PeerId, PeerInfo>,
     relay_index: BTreeSet<PeerId>,
     rendezvous_index: BTreeSet<PeerId>,
 }
 
 impl DiscoveryState {
     pub(crate) fn add_peer_addr(&mut self, peer_id: PeerId, addr: &Multiaddr) {
-        self.peers
+        let _ = self
+            .peers
             .entry(peer_id)
             .or_default()
             .addrs
             .insert(addr.clone());
     }
 
     pub(crate) fn remove_peer(&mut self, peer_id: &PeerId) {
-        self.peers.remove(peer_id);
-        self.relay_index.remove(peer_id);
-        self.rendezvous_index.remove(peer_id);
+        drop(self.peers.remove(peer_id));
+        let _ = self.relay_index.remove(peer_id);
+        let _ = self.rendezvous_index.remove(peer_id);
```

Comment:
other such cases

Reply:
See explanations and links to lints given above 🙂